rust/connlib/clients/shared/src/lib.rs

```diff
@@ -56,12 +48,13 @@ impl Session {
     /// This connects to the portal a specified using [`LoginUrl`] and creates a wireguard tunnel using the provided private key.
     pub fn connect<CB: Callbacks + 'static>(
         args: ConnectArgs<CB>,
+        portal: PhoenixChannel<(), IngressMessages, ReplyMessages>,
```

Comment:
👀 👀 
Is this the spot where I could inject fake portal messages and do tests with a full Client with real platform-specific I/O without needing to run a real Elixir portal?